aws/resource_aws_sns_topic_subscription.go

```diff
@@ -328,3 +328,18 @@ func obfuscateEndpoint(endpoint string) string {
 	}
 	return obfuscatedEndpoint
 }
+
+func snsSubscriptionAttributeUpdate(snsconn *sns.SNS, tfId string, tfName string, val string) error {
+	awsAttrName := SNSSubscriptionAttributeMap[tfName]
+	req := &sns.SetSubscriptionAttributesInput{
+		SubscriptionArn: aws.String(tfId),
+		AttributeName:   aws.String(awsAttrName),
+		AttributeValue:  aws.String(val),
+	}
+	_, err := snsconn.SetSubscriptionAttributes(req)
+
+	if err != nil {
+		return fmt.Errorf("Unable to set %s attribute on subscription", awsAttrName)
```

Comment:
We should definitely return the `err` string here too 😄 
```go
return fmt.Errorf("Unable to set %s attribute on subscription: %s", attributeName, err)
```

Otherwise you get this, which isn't so helpful:
```
=== RUN   TestAccAWSSNSTopicSubscription_deliveryPolicy
--- FAIL: TestAccAWSSNSTopicSubscription_deliveryPolicy (37.69s)
	testing.go:513: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_sns_topic_subscription.test_subscription: 1 error(s) occurred:

		* aws_sns_topic_subscription.test_subscription: Unable to set DeliveryPolicy attribute on subscription
```

Reply:
Count on this